AlicerceTests/ServiceLocatorTests.swift

```diff
@@ -116,13 +210,110 @@ class ServiceLocatorTests : XCTestCase {
             XCTAssertEqual(rStringService, testLazyServiceString())
             
         } catch let error {
-            XCTAssert(false, "💥: Register failed with error: \(error)")
+            XCTFail("💥: Register failed with error: \(error)")
         }
     }
+
+    // MARK: - Failure
+
+    func testLazyRegister_UsingTypeInference_ShouldFailWithDuplicateLazyService() {
+
+        let testLazyServiceString: () -> (String) = {
+            return "Test Register Service"
+        }
+
+        do {
+
+            // Register the first lazy service
+            try serviceLocator.register(testLazyServiceString)
+
+            // Register the same lazy service should fail with duplicateService
+            try serviceLocator.register(testLazyServiceString)
+
+        } catch ServiceLocatorError.duplicateLazyServiceInit(_) {
+            print("✅"); return
+        } catch {
+            XCTFail("💥: Lazy Register failed with error: \(error)")
+        }
+    }
+
+    func testLazyRegister_UsingName_ShouldFailWithDuplicateLazyService() {
+
+        let testLazyServiceString: () -> (String) = {
+            return "Test Register Service"
+        }
+        let serviceName = "TestServiceName"
+
+        do {
+
+            // Register the first lazy service using name
+            try serviceLocator.register(name: serviceName, testLazyServiceString)
+
+            // Register the same lazy service, using the same name, should fail with duplicateService
+            try serviceLocator.register(name: serviceName, testLazyServiceString)
+
+        } catch ServiceLocatorError.duplicateLazyServiceInit(_) {
+            print("✅")
+        } catch {
+            XCTFail("💥: Lazy Register failed with error: \(error)")
+        }
+    }
+
+    func testLazyRegisterWithNormal_UsingTypeInference_ShouldFailWithDuplicateService() {
```

Comment:
This name is a bit confusing to me, especially the `"WithNormal" part.

If I understand correctly, this case tests whether a service registered as lazy using type inference that then gets initialised (and thus "updated" to normal ) should fail with `duplicateService` (and not `duplicateLazyService`) when another register attempt is made with the same name.

If this is correct, why not call these tests something like:

`testLazyRegister_UsingTypeInferenceAndInitialisingIt_ShouldFailWithDuplicateService`

What do you say?